app/models/season/phase_switcher.rb

```diff
@@ -1,44 +1,61 @@
+# frozen_string_literal: true
+
 class Season::PhaseSwitcher
 
+  WHITELISTED_PHASES = %i[
+        fake_proposals_phase
+        fake_application_phase
+        fake_coding_phase
+        back_to_reality
+    ]
+
   def self.season
     Season.current
   end
 
+  def self.destined(phase: phase)
+    if phase.in?(WHITELISTED_PHASES)
```

Comment:
May slightly change if we opt into the raise-error-appraoch. But for now, again, just a community best practise, I'd vote for doing this in a single line instead, makes it a bit more expressive:
```ruby
self.public_send(phase) if phase.in?(PHASES)
```